lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -192,14 +198,11 @@ class TwilioConnectionTransport extends StateMachine {
       }
     });
 
-    if (options.eventObserver) {
-      options.eventObserver.setPublisher(this._eventPublisher);
-    }
-
+    this._eventObserver.setPublisher(eventPublisher);
```

Comment:
can we instead have caller ensure that publisher is already set before calling this class? That way both publisher and observer need not be sent down.

Reply:
@makarandp0 I can add a TODO for this, since making this change now will lead to more changes in other classes and test files, which I want to avoid.